File: src/lib/sample-content.ts
// SPDX-FileCopyrightText: 2026 compiledkernel-idk
//
// SPDX-License-Identifier: GPL-3.0-or-later

import type { Subject } from './store'

export const sampleSubjects: Subject[] = [
  {
    id: 'biologie',
    naam: 'Biologie',
    icon: '🧬',
    volgorde: 1,
    topics: [
      {
        id: 'fotosynthese',
        subjectId: 'biologie',
        titel: 'Fotosynthese',
        slug: 'fotosynthese',
        content: `# Fotosynthese

Fotosynthese is het proces waarbij planten **lichtenergie** omzetten in **chemische energie**. Dit is een van de belangrijkste processen op aarde, omdat het de basis vormt voor vrijwel alle voedselketens.

## Hoe werkt het?

Planten vangen zonlicht op met behulp van **chlorofyl**, het groene pigment in hun bladeren. Dit pigment bevindt zich in speciale celorganellen genaamd **chloroplasten**.

De basisformule van fotosynthese is:

> 6 CO₂ + 6 H₂O + lichtenergie → C₆H₁₂O₆ + 6 O₂

Dit betekent dat planten **koolstofdioxide** en **water** gebruiken, samen met licht, om **glucose** en **zuurstof** te maken.

## Belangrijke termen

- **Chlorofyl**: Het groene pigment dat licht absorbeert
- **Chloroplast**: Het celorganel waar fotosynthese plaatsvindt
- **Stomata**: Kleine openingen in bladeren voor gasuitwisseling
- **Glucose**: De suiker die als energiebron wordt geproduceerd
- **Xyleem**: Transportweefsel dat water naar de bladeren brengt

:::meerkeuze
vraag: Welk gas nemen planten op tijdens fotosynthese?
opties:
  - Zuurstof
  - Stikstof
  - Koolstofdioxide*
  - Waterstof
uitleg: Planten nemen CO₂ (koolstofdioxide) op uit de lucht en geven O₂ (zuurstof) af als bijproduct.
:::

:::invullen
vraag: Het groene pigment in bladeren heet ___
antwoord: chlorofyl
hint: Begint met 'ch' en geeft bladeren hun groene kleur
:::

:::waar-of-niet
vraag: Planten hebben alleen zonlicht nodig om te groeien.
antwoord: niet waar
uitleg: Planten hebben naast zonlicht ook water, koolstofdioxide en mineralen nodig om te groeien.
:::

:::koppelen
vraag: Koppel elk onderdeel aan zijn functie
paren:
  - [Chlorofyl, Absorbeert lichtenergie]
  - [Stomata, Gasuitwisseling]
  - [Xyleem, Watervervoer]
  - [Chloroplast, Plaats van fotosynthese]
:::

:::open
vraag: Leg in je eigen woorden uit waarom fotosynthese belangrijk is voor het leven op aarde.
kernwoorden: [zuurstof, voedsel, energie, CO₂, voedselketen]
:::

:::meerkeuze
vraag: Waar vindt fotosynthese plaats in een plantencel?
opties:
  - Mitochondriën
  - Celkern
  - Chloroplasten*
  - Ribosomen
uitleg: Chloroplasten zijn de celorganellen die chlorofyl bevatten en waar het proces van fotosynthese plaatsvindt.
:::
`,
      },
      {
        id: 'cellen',
        subjectId: 'biologie',
        titel: 'Cellen',
        slug: 'cellen',
        content: `# Cellen — De Bouwstenen van het Leven

Alle levende organismen bestaan uit **cellen**. Een cel is de kleinste eenheid van leven. Er zijn twee hoofdtypen cellen: **prokaryote** cellen (zoals bacteriën) en **eukaryote** cellen (zoals planten- en diercellen).

## Onderdelen van een diercel

| Organel | Functie |
|---------|---------|
| **Celkern** | Bevat het DNA, bestuurt de cel |
| **Celmembraan** | Beschermt de cel, regelt transport |
| **Cytoplasma** | Gelachtige vloeistof in de cel |
| **Mitochondriën** | Energieproductie (celademhaling) |
| **Ribosomen** | Maken eiwitten aan |

## Verschil planten- en diercel

Plantencellen hebben extra onderdelen die diercellen niet hebben:
- **Celwand**: Stevige buitenlaag van cellulose
- **Chloroplasten**: Voor fotosynthese
- **Grote vacuole**: Opslag van water en voedingsstoffen

:::meerkeuze
vraag: Welk organel wordt de 'energiecentrale' van de cel genoemd?
opties:
  - Ribosomen
  - Mitochondriën*
  - Celkern
  - Chloroplasten
uitleg: Mitochondriën zetten glucose om in energie (ATP) via celademhaling. Ze worden daarom de 'energiecentrales' van de cel genoemd.
:::

:::waar-of-niet
vraag: Zowel planten- als diercellen hebben een celwand.
antwoord: niet waar
uitleg: Alleen plantencellen hebben een celwand (gemaakt van cellulose). Diercellen hebben alleen een celmembraan.
:::

:::invullen
vraag: De gelachtige vloeistof in een cel heet het ___
antwoord: cytoplasma
hint: Begint met 'cyto'
:::

:::koppelen
vraag: Koppel het organel aan de juiste functie
paren:
  - [Celkern, Bestuurt de cel]
  - [Mitochondriën, Maakt energie]
  - [Ribosomen, Maakt eiwitten]
  - [Celmembraan, Beschermt de cel]
:::

:::meerkeuze
vraag: Wat heeft een plantencel dat een diercel NIET heeft?
opties:
  - Celmembraan
  - Celkern
  - Chloroplasten*
  - Mitochondriën
uitleg: Chloroplasten komen alleen voor in plantencellen. Ze bevatten chlorofyl en zijn nodig voor fotosynthese.
:::
`,
      },
      {
        id: 'ecosystemen',
        subjectId: 'biologie',
        titel: 'Ecosystemen',
        slug: 'ecosystemen',
        content: `# Ecosystemen

Een **ecosysteem** is een leefgemeenschap van organismen samen met hun niet-levende omgeving. Alle organismen in een ecosysteem zijn met elkaar verbonden door voedselrelaties.

## Voedselketen

Een voedselketen laat zien wie wat eet:

> Gras → Konijn → Vos → Arend

Elke schakel in de keten noemen we een **trofisch niveau**:
1. **Producenten**: Planten die energie maken via fotosynthese
2. **Primaire consumenten**: Herbivoren (planteneters)
3. **Secundaire consumenten**: Carnivoren (vleeseters)
4. **Tertiaire consumenten**: Toproofdieren
5. **Reducenten**: Breken dood materiaal af (schimmels, bacteriën)

## Voedselweb

In werkelijkheid eten de meeste dieren meer dan één soort voedsel. Daarom spreken we van een **voedselweb** — een netwerk van meerdere voedselketens.

:::meerkeuze
vraag: Wat is de rol van planten in een ecosysteem?
opties:
  - Reducenten
  - Consumenten
  - Producenten*
  - Afbrekers
uitleg: Planten zijn producenten — ze maken hun eigen voedsel via fotosynthese en vormen de basis van elke voedselketen.
:::

:::invullen
vraag: Een netwerk van meerdere voedselketens noemen we een ___
antwoord: voedselweb
hint: Twee woorden, begint met 'voedsel'
:::

:::waar-of-niet
vraag: Reducenten staan altijd aan het begin van een voedselketen.
antwoord: niet waar
uitleg: Reducenten (zoals schimmels en bacteriën) breken dood organisch materiaal af. Ze staan niet aan het begin maar spelen een rol bij het recyclen van voedingsstoffen.
:::

:::koppelen
vraag: Koppel het trofisch niveau aan het juiste voorbeeld
paren:
  - [Producent, Gras]
  - [Primaire consument, Konijn]
  - [Secundaire consument, Vos]
  - [Reducent, Schimmel]
:::

:::open
vraag: Wat zou er gebeuren als alle producenten uit een ecosysteem zouden verdwijnen?
kernwoorden: [voedselketen, energie, herbivoren, uitsterven, instorten]
:::
`,
      },
    ],
  },
]

File: src/models/sample_content.py
# SPDX-FileCopyrightText: 2026 compiledkernel-idk
#
# SPDX-License-Identifier: GPL-3.0-or-later

"""
sample_content.py - Sample Biologie (Biology) content in Dutch.

Provides realistic study content with embedded question blocks that can be
parsed by content_parser.parse_content().
"""

from __future__ import annotations


def get_sample_subjects() -> list[dict]:
    """Return a list of sample subject dicts with topics and embedded questions."""

    fotosynthese_content = """\
# Fotosynthese

## Wat is fotosynthese?

Fotosynthese is het proces waarbij **groene planten**, algen en sommige bacteriën \
lichtenergie omzetten in chemische energie. Dit proces vindt plaats in de \
**chloroplasten** van plantencellen, met name in het **bladgroenkorrel** (chlorofyl).

De algemene vergelijking van fotosynthese is:

**6 CO₂ + 6 H₂O + lichtenergie → C₆H₁₂O₆ + 6 O₂**

## De lichtreactie

De **lichtreactie** vindt plaats in de **thylakoïdmembranen** van de chloroplasten. \
Tijdens deze fase wordt lichtenergie opgevangen door chlorofyl en omgezet in \
chemische energie in de vorm van **ATP** en **NADPH**. Water wordt hierbij gesplitst \
(*fotolyse*), waarbij zuurstof vrijkomt als bijproduct.

:::meerkeuze
vraag: Waar vindt de lichtreactie van fotosynthese plaats?
opties: A) In de mitochondriën B) In de thylakoïdmembranen C) In het cytoplasma D) In de celkern
correct: 1
uitleg: De lichtreactie vindt plaats in de thylakoïdmembranen van de chloroplasten, waar chlorofyl het licht opvangt.
:::

## De donkerreactie (Calvincyclus)

De **donkerreactie**, ook wel de **Calvincyclus** genoemd, vindt plaats in het \
**stroma** van de chloroplasten. Ondanks de naam heeft deze reactie geen \
duisternis nodig — het betekent alleen dat licht niet *direct* nodig is.

Tijdens de Calvincyclus wordt **CO₂** vastgelegd en omgezet in **glucose** met \
behulp van de ATP en NADPH uit de lichtreactie.

:::invullen
vraag: De donkerreactie vindt plaats in het _____ van de chloroplasten.
antwoord: stroma
hint: Het is de vloeistof rondom de thylakoïden.
:::

## Factoren die fotosynthese beïnvloeden

Verschillende factoren beïnvloeden de snelheid van fotosynthese:

- **Lichtintensiteit**: meer licht = snellere fotosynthese (tot een maximum)
- **CO₂-concentratie**: meer CO₂ = snellere reactie (tot een maximum)
- **Temperatuur**: enzymen werken optimaal rond **25-35°C**
- **Watervoorziening**: water is een grondstof voor de reactie

:::waar-of-niet
vraag: Bij een hogere temperatuur verloopt fotosynthese altijd sneller.
antwoord: niet
uitleg: Boven de optimale temperatuur denatureren enzymen, waardoor de fotosynthese juist afneemt.
:::

## Belang van fotosynthese

Fotosynthese is essentieel voor het leven op aarde. Het zorgt voor:

1. **Zuurstofproductie** — nodig voor de ademhaling van de meeste organismen
2. **Voedselproductie** — glucose als energiebron voor de plant en andere organismen
3. **CO₂-opname** — helpt bij het reguleren van het broeikaseffect

:::koppelen
vraag: Koppel de juiste termen aan hun functie in fotosynthese.
paren: Chlorofyl = Vangt lichtenergie op, Stroma = Plaats van de Calvincyclus, Thylakoïd = Plaats van de lichtreactie, ATP = Energiedrager
:::

:::open
vraag: Leg uit waarom fotosynthese belangrijk is voor het leven op aarde. Noem minstens drie redenen.
kernwoorden: zuurstof, glucose, voedsel, CO₂, broeikaseffect, energie, ademhaling
:::
"""

    cellen_content = """\
# Cellen

## De bouwstenen van het leven

Alle levende organismen zijn opgebouwd uit **cellen**. De cel is de kleinste \
eenheid van leven die zelfstandig kan functioneren. Er zijn twee hoofdtypen cellen:

- **Prokaryote cellen** — geen celkern (bijv. bacteriën)
- **Eukaryote cellen** — wel een celkern (bijv. planten- en diercellen)

## De plantencel

Een plantencel bevat de volgende organellen:

| Organel | Functie |
|---------|---------|
| **Celwand** | Stevigheid en bescherming |
| **Celmembraan** | Regelt transport van stoffen |
| **Celkern** | Bevat DNA, stuurt celprocessen aan |
| **Chloroplasten** | Fotosynthese |
| **Mitochondriën** | Celademhaling (energieproductie) |
| **Vacuole** | Opslag van water en stoffen |
| **Endoplasmatisch reticulum** | Eiwitsynthese en transport |

:::meerkeuze
vraag: Welk organel is verantwoordelijk voor de celademhaling?
opties: A) Chloroplasten B) Celkern C) Mitochondriën D) Vacuole
correct: 2
uitleg: Mitochondriën zijn de 'energiecentrales' van de cel en zorgen voor de celademhaling waarbij glucose wordt omgezet in ATP.
:::

## Verschil tussen planten- en diercellen

Plantencellen hebben enkele onderdelen die diercellen **niet** hebben:

- **Celwand** — geeft extra stevigheid
- **Chloroplasten** — voor fotosynthese
- **Grote centrale vacuole** — voor wateropslag

Diercellen hebben daarentegen **lysosomen** die belangrijker zijn voor de \
afbraak van afvalstoffen.

:::waar-of-niet
vraag: Diercellen bevatten chloroplasten.
antwoord: niet
uitleg: Chloroplasten komen alleen voor in plantencellen (en sommige algen). Diercellen kunnen geen fotosynthese uitvoeren.
:::

## Celdeling: mitose

**Mitose** is het proces waarbij een cel zich deelt in twee identieke dochtercellen. \
Dit is belangrijk voor **groei**, **herstel** en **vervanging** van cellen.

De fasen van mitose zijn:

1. **Profase** — chromosomen worden zichtbaar, kernmembraan verdwijnt
2. **Metafase** — chromosomen liggen op de evenaar van de cel
3. **Anafase** — chromatiden worden naar de polen getrokken
4. **Telofase** — nieuwe kernmembranen vormen zich, cel splitst

:::invullen
vraag: Tijdens de _____ liggen de chromosomen op de evenaar van de cel.
antwoord: metafase
hint: Deze fase begint met de letter M en betekent letterlijk 'midden-fase'.
:::

:::koppelen
vraag: Koppel elke fase van mitose aan de juiste beschrijving.
paren: Profase = Chromosomen worden zichtbaar, Metafase = Chromosomen op de evenaar, Anafase = Chromatiden naar de polen, Telofase = Nieuwe kernmembranen vormen
:::

:::open
vraag: Beschrijf het verschil tussen een prokaryote en een eukaryote cel. Geef van beide een voorbeeld.
kernwoorden: celkern, DNA, bacterie, prokaryoot, eukaryoot, organellen, membraan
:::
"""

    ecosystemen_content = """\
# Ecosystemen

## Wat is een ecosysteem?

Een **ecosysteem** is een samenlevingsverband van **levende organismen** \
(biotische factoren) en hun **niet-levende omgeving** (abiotische factoren). \
Voorbeelden van ecosystemen zijn een bos, een meer, een woestijn of een koraalrif.

### Biotische factoren
- Planten, dieren, schimmels, bacteriën
- Onderlinge relaties: predatie, competitie, symbiose

### Abiotische factoren
- Temperatuur, licht, water, bodem, mineralen
- Bepalen welke organismen er kunnen leven

:::meerkeuze
vraag: Welke van de volgende is een abiotische factor?
opties: A) Een wolf B) Temperatuur C) Een eikenboom D) Bacteriën
correct: 1
uitleg: Temperatuur is een niet-levende (abiotische) factor. Wolven, eikenbomen en bacteriën zijn allemaal levende (biotische) factoren.
:::

## Voedselketens en voedselwebben

Energie stroomt door een ecosysteem via **voedselketens**:

**Producent → Primaire consument → Secundaire consument → Tertiaire consument**

Voorbeeld:
- **Gras** (producent) → **Konijn** (herbivoor) → **Vos** (carnivoor) → **Arend** (toppredator)

Meerdere voedselketens vormen samen een **voedselweb**. Bij elke stap gaat \
ongeveer **90%** van de energie verloren als warmte.

:::invullen
vraag: Bij elke stap in de voedselketen gaat ongeveer _____% van de energie verloren.
antwoord: 90
hint: Het is een hoog percentage — slechts een tiende gaat verder.
:::

## Kringlopen

Stoffen worden in een ecosysteem **hergebruikt** via kringlopen:

### De koolstofkringloop
1. Planten nemen **CO₂** op via fotosynthese
2. Dieren eten planten en ademen **CO₂** uit
3. Dode organismen worden afgebroken door **reducenten** (schimmels, bacteriën)
4. Fossiele brandstoffen geven bij verbranding CO₂ af

### De stikstofkringloop
- **Stikstofbindende bacteriën** zetten N₂ om in bruikbare stikstofverbindingen
- Planten nemen **nitraat** op via hun wortels
- Dieren krijgen stikstof via voedsel

:::waar-of-niet
vraag: Planten nemen stikstof rechtstreeks op als N₂ uit de lucht.
antwoord: niet
uitleg: Planten kunnen geen N₂ gas gebruiken. Stikstofbindende bacteriën moeten het eerst omzetten in nitraat of ammonium.
:::

## Verstoringen van ecosystemen

Menselijke activiteiten kunnen ecosystemen verstoren:

- **Ontbossing** — verlies van habitats en biodiversiteit
- **Vervuiling** — giftige stoffen in water, bodem en lucht
- **Klimaatverandering** — verschuiving van leefgebieden
- **Overbevissing** — verstoring van voedselwebben

:::koppelen
vraag: Koppel de verstoring aan het gevolg.
paren: Ontbossing = Verlies van habitats, Vervuiling = Giftige stoffen in het milieu, Klimaatverandering = Verschuiving van leefgebieden, Overbevissing = Verstoring van voedselwebben
:::

:::open
vraag: Leg uit hoe de koolstofkringloop werkt en welke rol fotosynthese en ademhaling daarin spelen.
kernwoorden: CO₂, fotosynthese, ademhaling, planten, dieren, reducenten, kringloop, glucose
:::
"""

    biologie_subject = {
        "id": "bio-001",
        "naam": "Biologie",
        "icon": "\U0001f9ec",  # 🧬
        "volgorde": 0,
        "topics": [
            {
                "id": "topic-bio-001",
                "subjectId": "bio-001",
                "titel": "Fotosynthese",
                "content": fotosynthese_content,
                "slug": "fotosynthese",
            },
            {
                "id": "topic-bio-002",
                "subjectId": "bio-001",
                "titel": "Cellen",
                "content": cellen_content,
                "slug": "cellen",
            },
            {
                "id": "topic-bio-003",
                "subjectId": "bio-001",
                "titel": "Ecosystemen",
                "content": ecosystemen_content,
                "slug": "ecosystemen",
            },
        ],
    }

    return [biologie_subject]
